tests/serial/time/implicit/parallel-coupling/ReadWriteScalarDataWithSubcycling.cpp

```diff
@@ -0,0 +1,142 @@
+  double currentDt       = dt > maxDt ? maxDt : dt;                      // determine actual timestep length; must fit into remaining time in window
+  double time            = timestep * dt;
+
+  if (precice.isActionRequired(precice::constants::actionWriteInitialData())) {
+    writeData = writeFunction(time);
+    precice.writeScalarData(writeDataID, vertexID, writeData);
+    precice.markActionFulfilled(precice::constants::actionWriteInitialData());
+  }
+
+  precice.initializeData();
+
+  while (precice.isCouplingOngoing()) {
+    if (precice.isActionRequired(precice::constants::actionWriteIterationCheckpoint())) {
+      windowStartTime = time;
+      windowStartStep = timestep;
+      precice.markActionFulfilled(precice::constants::actionWriteIterationCheckpoint());
+    }
+
+    if (timestep % nSubsteps == 0) {
+      BOOST_TEST(precice.isReadDataAvailable());
+    } else {
+      BOOST_TEST(!precice.isReadDataAvailable());
+    }
+
+    if (precice.isReadDataAvailable()) {
+      precice.readScalarData(readDataID, vertexID, readData);
+    }
+    if (iterations == 0 && timestep == 0) {                                    // special situation: Both solvers are in their very first time windows, first iteration, first time step
+      BOOST_TEST(readData == readFunction(startTime));                         // use initial data only.
+    } else if (iterations == 0) {                                              // special situation: Both solvers get the old data for all time windows.
+      BOOST_TEST(readData == readFunction(startTime + timewindow * windowDt)); // data at end of window was written by other solver.
+    } else if (iterations > 0) {
+      BOOST_TEST(readData == readFunction(startTime + (timewindow + 1) * windowDt));
+    } else { // we should not enter this branch, because this would skip all tests.
+      BOOST_TEST(false);
+    }
+
+    // solve usually goes here. Dummy solve: Just sampling the writeFunction.
+    BOOST_TEST(currentDt == expectedDts[timestep % nSubsteps]);
+    time += currentDt;
+    if (precice.isWriteDataRequired(currentDt)) {
+      writeData = writeFunction(time);
+      precice.writeScalarData(writeDataID, vertexID, writeData);
+    }
+    maxDt     = precice.advance(currentDt);
+    currentDt = dt > maxDt ? maxDt : dt;
+    timestep++;
+    if (precice.isActionRequired(precice::constants::actionReadIterationCheckpoint())) { // at end of window and we have to repeat it.
+      iterations++;
+      timestep = windowStartStep;
+      time     = windowStartTime;
+      precice.markActionFulfilled(precice::constants::actionReadIterationCheckpoint()); // this test does not care about checkpointing, but we have to make the action
+    }
+    if (precice.isTimeWindowComplete()) {
+      timewindow++;
+      iterations = 0;
```

Comment:
You could also test the value of `iterations` here

Reply:
Done in [c0550e4](https://github.com/precice/precice/pull/1186/commits/c0550e4ace598f11dab8d929dca8e4f90e34c832).